lib/first_last_success_calculator.rb

```diff
@@ -0,0 +1,143 @@
+# coding: utf-8
+#
+# Process the list of successfully accessed urls for each day to produce a
+# single file containing the dates when each such url was first and last
+# accessed.
+
+require 'open3'
+require_relative 'config_logging'
+
+class FirstLastSuccessCalculator
+  attr_reader :daily_successes_dir
+  attr_reader :first_last_file
+  attr_reader :first_last_sources_file
+
+  def initialize(processed_dir)
+    successes_dir = "#{processed_dir}/successes"
+    @daily_successes_dir = "#{successes_dir}/daily"
+    @first_last_file = "#{successes_dir}/first_last"
+    @first_last_sources_file = "#{successes_dir}/first_last_sources"
+  end
+
+  def process
+    remove_out_of_date_output
+    update_first_last
+  end
+
+private
+
+  def current_daily_files_and_sizes
+    @current_daily_files_and_sizes ||=
```

Comment:
Matybe having these on the same line could be confusing, perhaps wrapping the code in `begin` & `end` would help.